feat: Add chatgpt and Agents SDK integration (#2065)
apify/apify-docs: sources/platform/integrations/ai/openai_agents.md

---
title: OpenAI Agents SDK integration
sidebar_label: OpenAI Agents SDK
description: Learn how to integrate Apify with the OpenAI Agents SDK using the Model Context Protocol to build AI agents with web data access.
sidebar_position: 13
slug: /integrations/openai-agents
---

The _OpenAI Agents Python SDK_ enables you to build AI agents powered by OpenAI's language models that can use tools, manage context, and interact with external systems through the [Model Context Protocol (MCP)](https://openai.com/research/model-context-protocol). By connecting to the Apify MCP server, your agents can access Apify's extensive library of Actors to perform web scraping, data extraction, and automation tasks in real time.

For more details about the OpenAI Agents SDK, see the [official documentation](https://openai.github.io/openai-agents-python/).

## Prerequisites

Before integrating Apify with OpenAI Agents SDK, you'll need:

- _An Apify account_ - Sign up for an Apify account, if you don't have one.

> - _Apify API token_ - Get your API token from the **Integrations** section in [Apify Console](https://console.apify.com/account#/integrations). This token authorizes the MCP server to run Actors on your behalf. Make sure to keep it secure.

Reviewer: ```suggestion
- _Apify API token_ - Get your API token from the **Integrations** section in [Apify Console](https://console.apify.com/account#/integrations). This token authorizes the Apify MCP server to run Actors on your behalf. Make sure to keep it secure.
```

    # Create MCP server connection with Bearer token
    async with MCPServerStreamableHttp(
        name="Apify MCP Server",
        params={
            "url": "https://mcp.apify.com",
            "headers": {"Authorization": f"Bearer {os.environ['APIFY_TOKEN']}"},
            "timeout": 120,
        },
        cache_tools_list=True,
        max_retry_attempts=3,
    ) as server:
        # Create agent with MCP server
        agent = Agent(
            name="Assistant",
            instructions="Use the Apify MCP tools to answer questions. Search the web when needed.",
            mcp_servers=[server],
        )

        # Run the agent
        result = await Runner.run(agent, "Search the web and summarize recent trends in AI agents")
        print(result.final_output)

if __name__ == "__main__":
    asyncio.run(main())
```

#### Configuration options

The `MCPServerStreamableHttp` connects to the Apify MCP server using streamable HTTP. Key configuration options:

- `params`: Dictionary containing:
  - `url`: The MCP server URL (`https://mcp.apify.com` for Apify)
  - `headers`: Authentication headers (Bearer token for Apify)
  - `timeout`: Request timeout in seconds

> - `cache_tools_list`: Caches the tool list to reduce API calls

Reviewer: Is this a boolean or a list of tools to be cached? I assume the former.
```suggestion
- `cache_tools_list`: A boolean property that specifies whether to cache the tool list to reduce API calls
```

```python
import asyncio
import os

from agents import Agent, Runner
from agents.mcp import MCPServerStreamableHttp

os.environ["APIFY_TOKEN"] = "Your Apify API token"
os.environ["OPENAI_API_KEY"] = "Your OpenAI API key"

async def main() -> None:
    # Create MCP server connection with Instagram scraper
    async with MCPServerStreamableHttp(
        name="Apify MCP Server",
        params={
            "url": "https://mcp.apify.com?tools=apify/instagram-scraper",
            "headers": {"Authorization": f"Bearer {os.environ['APIFY_TOKEN']}"},
            "timeout": 120,
        },
        cache_tools_list=True,
        max_retry_attempts=3,
    ) as server:
        # Create agent with MCP server
        agent = Agent(
            name="Assistant",
            instructions="Use the Instagram scraper to analyze Instagram profiles and posts.",
            mcp_servers=[server],
        )

        # Run the agent
        result = await Runner.run(
            agent, "Get the latest posts from @natgeo Instagram profile and summarize the main themes"
        )
        print(result.final_output)

if __name__ == "__main__":
    asyncio.run(main())
```

> ### Agent to test MCP servers

Reviewer: ```suggestion
### MCP server testing Agent
```

## Troubleshooting

### Authentication errors

- _Check your API token_: Verify that your Apify API token is correct. You can find it in the **Integrations** section of the [Apify Console](https://console.apify.com/account#/integrations). Without a valid token, the server cannot start Actor runs.
- _Set environment variables_: When running your agent, ensure you have set the `APIFY_TOKEN` and `OPENAI_API_KEY` environment variables.

### Agent execution issues

- _No response or long delays_: Actor runs can take time to complete depending on their task. If you're experiencing long delays, check the Actor's logs in Apify Console. The logs will provide insight into the Actor's status and show if it's processing a long operation or has encountered an error.
- _Tool execution timeout_: If tool calls are timing out, increase the `timeout` parameter in the `MCPServerStreamableHttp` configuration.

## Related integrations

- [ChatGPT integration](/platform/integrations/chatgpt) — Add Apify MCP server as a custom connector in ChatGPT

> - [OpenAI Assistants integration](/platform/integrations/openai-assistants) — Use Apify Actors with OpenAI Assistants API via function calling

Reviewer: ```suggestion
- [ChatGPT integration](/platform/integrations/chatgpt) - Add Apify MCP server as a custom connector in ChatGPT
- [OpenAI Assistants integration](/platform/integrations/openai-assistants) - Use Apify Actors with OpenAI Assistants API via function calling
```
